packages/shared/components/Notification/Notification.tsx

```diff
@@ -1,51 +1,49 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 import React, { useEffect, useRef, useState } from 'react';
 import styled, { css, useTheme } from 'styled-components';
 import { ButtonIcon, Flex, Text } from 'design';
-import { Close, Info, Warning } from 'design/Icon';
+import { Close } from 'design/Icon';
+import { propTypes } from 'design/system';
 
 import { NotificationItem, NotificationItemContent } from './types';
 
 interface NotificationProps {
   item: NotificationItem;
-
   onRemove(): void;
+  Icon: React.ElementType;
+  getColor(theme): string;
+  isAutoRemovable: boolean;
+  autoRemoveDurationMs?: number;
 }
 
-const notificationConfig: Record<
-  NotificationItem['severity'],
-  { Icon: React.ElementType; getColor(theme): string; isAutoRemovable: boolean }
-> = {
-  error: {
-    Icon: Warning,
-    getColor: theme => theme.colors.danger,
-    isAutoRemovable: false,
-  },
-  warn: {
-    Icon: Warning,
-    getColor: theme => theme.colors.warning,
-    isAutoRemovable: true,
-  },
-  info: {
-    Icon: Info,
-    getColor: theme => theme.colors.info,
-    isAutoRemovable: true,
-  },
-};
-
-const autoRemoveDurationMs = 10_000; // 10s
-
-export function Notification(props: NotificationProps) {
+const defaultAutoRemoveDurationMs = 10_000; // 10s
+
+export function Notification(props: NotificationProps & propTypes) {
```

Comment:
Since this component is now in the shared package, I think it'd be worthwhile to add some stories for it.

There's already some stories for the bigger `Notifications` component in Connect but I think here we could have one simple notification and one with `isAutoRemovable` set to `true`. Just simple enough so that if someone wants to use this shared component, they'll see that there are stories for it.